targets/TARGET_STM/i2c_api.c

```diff
@@ -296,7 +296,7 @@ void i2c_frequency(i2c_t *obj, int hz)
     struct i2c_s *obj_s = I2C_S(obj);
     I2C_HandleTypeDef *handle = &(obj_s->handle);
 
-    MBED_ASSERT((hz > 0) && (hz <= 400000));
+    MBED_ASSERT((hz == 100000) || (hz == 400000) || (hz == 1000000));
```

Comment:
Those are I2C supported frequencies 
standard mode: 100 kbit/s, full speed: 400 kbit/s, fast mode: 1 mbit/s
Specific timings must be calculated for each frequency, and only timings for those values are implemented. Other requested frequencies may result in undefined behavior.